### ISSUE TITLE
roundf is not included in math.h for Visual Studio 2010

### ISSUE THREAD

**Issue by anthrotype** [adobe-type-tools/afdko#56]
although part of C99 standard, the `roundf` function was only included in MS Visual Studio only from version 2013:
http://blogs.msdn.com/b/vcblog/archive/2013/07/19/c99-library-support-in-visual-studio-2013.aspx

When I try to build the AFKDO using Visual Studio 2010, I get the following error:
```
"C:\UsersLocal\cosimo.lupo\Documents\GitHub\afdko\FDK\Tools\Programs\tx\build\win\vc10\tx.sln" (Build target) (1) -> "C:\UsersLocal\cosimo.lupo\Documents\GitHub\afdko\FDK\Tools\Programs\tx\build\win\vc10\tx.vcxproj" (default target)
(2) ->
(Link target) ->
  absfont.lib(absfont_dump.obj) : error LNK2019: unresolved external symbol _roundf referenced in function _writeRea l [C:\UsersLocal\cosimo.lupo\Documents\GitHub\afdko\FDK\Tools\Programs\tx\build\win\vc10\tx.vcxproj]
  svgwrite.lib(svgwrite.obj) : error LNK2001: unresolved external symbol _roundf [C:\UsersLocal\cosimo.lupo\Document s\GitHub\afdko\FDK\Tools\Programs\tx\build\win\vc10\tx.vcxproj]
  ufowrite.lib(ufowrite.obj) : error LNK2001: unresolved external symbol _roundf [C:\UsersLocal\cosimo.lupo\Document s\GitHub\afdko\FDK\Tools\Programs\tx\build\win\vc10\tx.vcxproj]
  ..\..\..\exe\win\release\tx.exe : fatal error LNK1120: 1 unresolved externals [C:\UsersLocal\cosimo.lupo\Documents \GitHub\afdko\FDK\Tools\Programs\tx\build\win\vc10\tx.vcxproj]
```

There are three modules where the roundf is used, all in the lib: `absfont_dump.c`, `svgwrite.c` and `ufowrite.c`. Because of the unresolved external symbol, the tools `tx`, `mergeFonts` and `rotateFont` cannot be successfully built on Windows using vc10.
The *.sln files version read "Visual Studio 2010", and also the name of the folders for the Windows build is "vc10" (i.e. MS Visual Studio 2010). So one would expect to compile the AFDKO code with Visual Studio 2010.

If I define the `roundf` function like below, then I can successfully compile everything:

```c
/* roundf was only included in math.h as of MS Visual Studio 2013, so we must define our own. */
#if _MSC_VER < 1800
static float roundf(float num)
{
	float integer = ceilf(num);
	if (num > 0)
		return integer - num > 0.5f ? integer - 1.0f : integer;
	return integer - num >= 0.5f ? integer - 1.0f : integer;
}
#endif
```

(I found the code for roundf [here](http://www.opensource.apple.com/source/JavaScriptCore/JavaScriptCore-721.26/wtf/MathExtras.h), but there's nothing special about it).

However, the organization of the AFDKO sources is very complex and I don't know where to place this function so that I can include it in all the required modules.

Could anybody help me out?

**Comment by anthrotype** [comment]
I just noticed that the use of roundf function was introduced with this commit https://github.com/adobe-type-tools/afdko/commit/aeb21404f1729c3f68e31497c92e4fe5f59d823f

**Comment by readroberts** [comment]
Right. I am in the process of fixing this, along with some other bugs. New checkin should by by Monday, maybe late today.

**Comment by anthrotype** [comment]
thank you!

**Comment by anthrotype** [comment]
I see you have fixed this issue with commit dffc4d3fe2d2478b266ecb4b91e845352039c273
thanks a lot!
